orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltOrchestrator.java

```diff
@@ -489,22 +497,25 @@ public void changePrimaryGateway(GatewayConfig formerGateway, GatewayConfig newP
     @Override
     public void installFreeIPA(GatewayConfig primaryGateway, List<GatewayConfig> allGatewayConfigs, Set<Node> allNodes,
             ExitCriteriaModel exitCriteriaModel) throws CloudbreakOrchestratorException {
-        Set<String> primaryServerHostname = Collections.singleton(primaryGateway.getHostname());
+        Set<String> freeIpaMasterHostname = new HashSet<>(getHostnamesForRoles(primaryGateway, Set.of(FREEIPA_MASTER_ROLE, FREEIPA_MASTER_REPLACEMENT_ROLE)));
+        Set<String> existingFreeIpaReplaceHostnames = new HashSet<>(getHostnamesForRoles(primaryGateway, Set.of(FREEIPA_REPLICA_ROLE)));
 
-        Set<String> replicaServersHostnames = allGatewayConfigs.stream()
-                .filter(gwc -> !gwc.getHostname().equals(primaryGateway.getHostname()))
-                .map(GatewayConfig::getHostname).collect(Collectors.toSet());
+        Set<String> unassignedHostnames = allGatewayConfigs.stream()
+                .map(GatewayConfig::getHostname)
+                .filter(hostname -> !freeIpaMasterHostname.contains(hostname))
+                .filter(hostname -> !existingFreeIpaReplaceHostnames.contains(hostname))
+                .collect(Collectors.toCollection(HashSet::new));
 
         try (SaltConnector sc = createSaltConnector(primaryGateway, saltErrorResolver)) {
-            LOGGER.debug("Set primary FreeIPA: {}", primaryServerHostname);
-            saltCommandRunner.runSaltCommand(sc, new GrainAddRunner(primaryServerHostname, allNodes, "freeipa_primary"), exitCriteriaModel, exitCriteria);
-            runNewService(sc, new HighStateRunner(primaryServerHostname, allNodes), exitCriteriaModel);
-
-            LOGGER.debug("Set replica FreeIPA: {}", replicaServersHostnames);
-            if (!replicaServersHostnames.isEmpty()) {
-                saltCommandRunner.runSaltCommand(sc, new GrainAddRunner(replicaServersHostnames, allNodes, "freeipa_replica"), exitCriteriaModel, exitCriteria);
-                runNewService(sc, new HighStateRunner(replicaServersHostnames, allNodes), exitCriteriaModel);
-            }
+
+            installFreeIPAUpdateExistingReplicas(sc, existingFreeIpaReplaceHostnames, allNodes, exitCriteriaModel);
```

Comment:
so I'm just dropping it here, but I noticed somewhere else also. We used everywhere `FreeIpa` until now (at least I hope so), and not `FreeIPA`. Could you rename methods, classes to match this?

Reply:
I fixed this one....but I will put the others in a separate PR. There are quite a few capitalization changes (19 files and their associated classes).